build: add appropriate compiler flags

lgray opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

This adds back compiler flags that appear in `getopt.vcxproj` that are not default in cmake. Though it doesn't seem to have broken anything.

lgray commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15795834535](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15795834535). Examine the logs at this URL for more detail.</sub>

lgray commented:
@nehaljwani OK - silenced some compiler warnings and now using the configuration in the original solution file (as far as I can tell). Please go ahead and merge when you're happy with it.

lgray commented:
@nehaljwani ok - now properly done. Wanted to make some things even between vcpkg and conda in terms of naming as well. Please merge when you have time.

lgray commented:
And again, apologies for the noise, it is difficult to catch this stuff when there are no tests.

lgray commented:
and @hmaarrfk just so you're in the loop (since you're the only other person who has contributed to this repository)

nehaljwani commented on recipe/0001-inject-cmake-project.patch:
```diff
@@ -64,6 +64,9 @@ index 0000000..c15df47
 +                            $<INSTALL_INTERFACE:include>
 +                           )
 +
++target_compile_definitions(getopt PRIVATE EXPORTS_GETOPT _CONSOLE)
```
What is `_CONSOLE` for?

lgray replied:
I actually don't know. I am merely emulating the behavior over in the original getopt.vcxproj https://github.com/libimobiledevice-win32/getopt/blob/master/getopt.vcxproj#L83.

Since this switches away from that and I am unfamiliar with this code, I am happy to forward what was there before.

nehaljwani replied:
Please add a comment explaining this at the top of the file with a link.

lgray replied:
done

lgray replied:
Following up, `_CONSOLE` appears to be a MSVC convention for labelling programs as commandline oriented. The definition itself doesn't appear in `getopt.h` or `getopt.c` but may appear in the MSVC implementations of the c standard libraries, so best to leave it around.

nehaljwani commented on recipe/0001-inject-cmake-project.patch:
```diff
@@ -64,6 +64,9 @@ index 0000000..c15df47
 +                           )
 +
++target_compile_definitions(getopt PRIVATE EXPORTS_GETOPT _CONSOLE)
++target_compile_options(getopt PRIVATE /Oi /O2 /W3 /Gy)
```
Why do you need /W3?

Why not let CMake control /O2 with the build type?

lgray replied:
Another emulation of the flags in https://github.com/libimobiledevice-win32/getopt/blob/master/getopt.vcxproj#L88 in both cases https://github.com/libimobiledevice-win32/getopt/blob/master/getopt.vcxproj#L81

lgray replied:
done

nehaljwani commented on recipe/0001-inject-cmake-project.patch:
```diff
@@ -88,34 +91,34 @@ index 0000000..c15df47
-+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/getopt
++  ${CMAKE_CURRENT_BINARY_DIR}/unofficial-getopt-win32Config.cmake
++  ${CMAKE_CURRENT_BINARY_DIR}/unofficial-getopt-win32ConfigVersion.cmake
++  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/unofficial-getopt-win32
```
Why use the term 'unofficial' instead of maybe 'conda-forge'?

lgray replied:
This is emulating the behavior of the package when it gets installed with vcpkg. Where you `find_package(unofficial-getopt-win32)`, and people always appreciate consistency, even if it is a bit obtuse.

lgray replied:
I'd be happy to change it to `conda-forge` but there is precedence for the `unofficial` name. Let me know your decision.

nehaljwani replied:
Please add a comment explaining this at the top of the file.

lgray replied:
done

lgray commented:
@nehaljwani thank you for your comments!

lgray committed:
appropriate compiler flags and naming improvements

lgray commented:
OK - should be all addressed now. Please let me know if you need anything else.

lgray commented:
@nehaljwani any further review, or can we go ahead and merge?

lgray commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15828983197](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15828983197). Examine the logs at this URL for more detail.</sub>

lgray committed:
Library/include not just include

lgray committed:
no - reading a bit more it should be only include

INSTALL_INTERFACE should add in anything relative to CMAKE_INSTALL_PREFIX which for windows conda builds is `%PREFIX%/Library`
